### DISCUSSION TITLE
ble.sh hides error output from some commands

### DISCUSSION THREAD

**Discussion by JohEngstrom** [akinomyoga/ble.sh#491] [Q&A]
I've recently found ble.sh while changing from zsh to bash. It's working great apart from one thing I've found.

For example when using git to push a new branch to remote you get this 'help' message:

```
fatal: The current branch demo has no upstream branch.
To push the current branch and set the remote as upstream, use

    git push --set-upstream origin demo
```

When I have ble.sh activated I just get `[ble: exit 128]`.

If I disable the ble.sh error with `bleopt exec_errexit_mark=''` then I get nothing apart from the caret turning red.

If there any way of letting errors show as normal?

Thanks!

**Comment by akinomyoga** [comment]
Does that happen only with the git command?

If that happens for all the commands that output error messages, one possibility is that you are redirecting `stderr` to somewhere different when performing `ble-attach`. How do you set up your `~/.bashrc` for `ble.sh`? Could you provide the content of your `~/.bashrc`?

**Reply by JohEngstrom** [reply]
Yes it's the same with all commands that output an error.

I'm using Ubuntu 24.04 in WSL2.

What I've found is that it works when I first open the prompt. Then I can see stderr output. If I then do a `source ~/.bashrc` then stderr is 'muted'. See screenshot below.

So I commented out everything apart from ble.sh in my .bashrc but I get the same behavior.

![blesh](https://github.com/user-attachments/assets/c3e66d0a-e236-4f05-8d73-ff3e58c3d62c)

My .bashrc looks like this now:

```
# Init ble.sh
[[ $- == *i* ]] && source ~/.local/share/blesh/ble.sh --noattach

# Init bash
# source ~/.shellConfig/bash/init.sh

# Environment variables
# source ~/.shellConfig/bash/env/envars.sh

# Aliases
# source ~/.shellConfig/bash/alias/alias.sh

# Functions
# source ~/.shellConfig/bash/functions/functions.sh

# Prompt - Starship, make sure to keep at the end of .bashrc
# source ~/.shellConfig/bash/prompt/starship.sh

# ble.sh | should always come last
[[ ${BLE_VERSION-} ]] && ble-attach
```

**Reply by akinomyoga** [reply]
Thank you for the details. I confirmed the behavior in my environment. I'll fix it.

**Reply by akinomyoga** [reply]
@JohEngstrom I pushed fixes 49f976188ad7f708c8f9c1423e54381b96b1863a and 670c7ea0d9f2e4debbc4021b0b8502a36b0b1999 in the `master` branch. Could you update ble.sh by running `ble-update` inside a ble.sh session and check if the problem is fixed in a new session?

**Reply by JohEngstrom** [reply]
Works perfectly now! Thanks a lot for fixing it so quickly 👍

**Reply by akinomyoga** [reply]
Thank you for checking!
